src/commands/update.ts

```diff
@@ -0,0 +1,41 @@
+import { SlashCommandBuilder } from 'discord.js';
+import { exec } from 'node:child_process';
+
+const info = new SlashCommandBuilder()
+	.setName('update')
+	.setDescription('Pulls the latest changes and restarts the bot')
+	.setDefaultMemberPermissions('0');
+
+export const update: GlobalCommand = {
+	info,
+	requiresGuild: false,
+	async execute({ replyPrivately, user, interaction }) {
+		const admin_ids = process.env['ADMINISTRATORS']?.split(',');
+		if (!admin_ids) {
+			// TODO: make this a UserMessageException
+			throw new Error('There is no ADMINISTRATORS variable. You must set ADMINISTRATORS in .env');
+		}
+
+		if (!admin_ids.includes(user.id)) {
+			// TODO: make this a UserMessageException
+			throw new Error(
+				'You do not have permission to perform this command. Contact the bot administrator.'
+			);
+		}
+
+		await replyPrivately('Updating...');
+		exec('npm run update', err => {
+			void (async (): Promise<void> => {
+				if (err) {
+					throw err;
```

Comment:
Since we invoke the function using `void`, these thrown errors are never caught. [`void`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void) only evaluates the given expression (in this case, kicking off a `Promise`-returning anonymous function) and returns `undefined` synchronously. If an error is thrown in the process of some asynchronous task that isn't `await`-ed, that error only ever appears in the console, and cannot be handled appropriately. Might even cause a crash, depending on runtime flags.

Reply:
obviated by the promisification